### PR TITLE
Adds Missing Sync Tests

### LINKED ISSUE #7: Add Tests For Sync Functionalty
the latest upstream changes didn't include tests

### PR DESCRIPTION
**Pull request by bonedaddy**
Adds missing tests for sync settings, closes https://github.com/RTradeLtd/go-ds-badger/issues/7

### CODE REVIEW

File: ds_test.go [hunk 1]
```diff
@@ -48,6 +49,23 @@ func newDS(t *testing.T) (*Datastore, func()) {
 	}
 }
 
+func newDSSync(t *testing.T, sync bool) (*Datastore, func()) {
+	path, err := ioutil.TempDir(os.TempDir(), "testing_badger_")
+	if err != nil {
+		t.Fatal(err)
+	}
+	opts := &DefaultOptions
+	opts.SyncWrites = sync
```
Comment (xiegeo) [inline ds_test.go:58]: Doesn't this change the default options for everything?

Comment (bonedaddy) [reply ds_test.go:58]: Not sure what you mean?

Comment (xiegeo) [reply ds_test.go:58]: DefaultOptions is get by reference, then changed.

**Review by xiegeo — APPROVED** [review]

File: ds_test.go (hunk 1 above)
Comment (bonedaddy) [reply ds_test.go:58]: Hmm I see, I don't think so winc this is instantiating a new default options instance and a reference to that yea?

Comment (bonedaddy) [reply ds_test.go:58]: > DefaultOptions is get by reference, then changed.

Actually my previous comment is incorrect, you are quite correct and i have pushed a commit fixing this.

**Pull request merged**